### PR TITLE
Criação de funcionalidade de inserção em bulk.

### PR DESCRIPTION
**Pull request by diogenesjcsilva**
Adicionado mapping para rota de bulk, apenas para inserts.
Adicionada função de bulk insert no controlador.
Adicionada funções de build de instancias antes de iniciar o bulk save e função de bulk save.

### CODE REVIEW

File: Platform.App/python-template/app/blueprints/mapping.py [hunk 1]
```diff
@@ -80,6 +80,20 @@ def persist_map(app_id):
...
+@mapping.route("/<app_id>/insert_bulk", methods=['POST'])
+def insert_bulk_map(app_id):
+    """ Insert data in bulk on domain """
+    instance_id = request.headers.get('Instance-Id')
+    process_id = request.headers.get('Process-Id')
+    reference_date = request.headers.get('Reference-Date')
+    body = json.loads(request.data)
+    controller = CommandController(app_id, body, instance_id, reference_date, process_id)
+    try:
+    return jsonify(controller.persist_bulk())
```
Comment (fabiogibson) [inline Platform.App/python-template/app/blueprints/mapping.py]: Code needs indentation.

File: Platform.App/python-template/core/temporal/listeners.py [hunk 2]
```diff
@@ -49,3 +49,23 @@ def before_flush(session, flush_context, instances):
...
+            entity, field, clock,None)
+            temporal_structures_history.append(history)
+
+        clock.ticks = 1
+        temporal_structures_clock.append(clock)
+
+    session.bulk_save_objects(temporal_structures_clock)
+    session.commit()
+    session.bulk_save_objects(temporal_structures_history)
+    session.commit()
```
Comment (fabiogibson) [inline Platform.App/python-template/core/temporal/listeners.py]: Please commit just once.

File: Platform.App/python-template/core/temporal/listeners.py [hunk 3]
```diff
@@ -49,3 +49,23 @@ def before_flush(session, flush_context, instances):
...
+        for field in entity.Temporal.fields:
+            history, history_created = session.create_field_history_bulk(
+            entity, field, clock,None)
+            temporal_structures_history.append(history)
+
+        clock.ticks = 1
+        temporal_structures_clock.append(clock)
+
+    session.bulk_save_objects(temporal_structures_clock)
+    session.commit()
```
Comment (fabiogibson) [inline Platform.App/python-template/core/temporal/listeners.py]: Please commit just once

File: Platform.App/python-template/model/persistence.py [hunk 4]
```diff
@@ -78,7 +78,19 @@ def persist(self, objs,scope):
         result += list(self.recover(to_recover,scope))
         self.destroy(to_destroy,scope)
         return result
+		
+    def persist_bulk(self, objs,scope):
+        self.link_branch(objs)
+        to_create = []
+        for o in objs:
+            if not self.is_valid_changed_obj(o):
```
Comment (fabiogibson) [inline Platform.App/python-template/model/persistence.py]: This check really needs to be made?

File: Platform.App/python-template/model/persistence.py [hunk 5]
```diff
@@ -93,6 +105,19 @@ def create(self, objs):
             self.session.add(instance)
             yield instance
 
+    def build_instance_bulk_create(self, objs):
+        for o in objs:
+            _type = o["_metadata"]["type"].lower()
+            #rid = o["_metadata"].get("rid",None)
```
Comment (fabiogibson) [inline Platform.App/python-template/model/persistence.py]: Please remove commented code

File: Platform.App/python-template/core/temporal/listeners.py [hunk 6]
```diff
@@ -49,3 +49,23 @@ def before_flush(session, flush_context, instances):
         temporal_structures.append(clock)
 
     session.add_all(temporal_structures)
+
+def addTemporal(session,instances):
```
Comment (fabiogibson) [inline Platform.App/python-template/core/temporal/listeners.py]: Please move this helper to utils.py

File: Platform.App/python-template/core/temporal/session.py [hunk 7]
```diff
@@ -20,6 +20,17 @@ def create_clock(self, entity, period=effective_now()):
...
+    
+    def create_clock_bulk(self, entity, period=effective_now()):
+        """creates a new clock for a temporal model.
+        """
+        clock = entity._clock()
+        clock.rid = uuid4()
+        clock.effective = period
+        clock.entity_id = entity.rid
+        clock.ticks = 0
+        #  self.add(clock)
```
Comment (fabiogibson) [inline Platform.App/python-template/core/temporal/session.py]: Please remove commented code.

File: Platform.App/python-template/core/temporal/session.py [hunk 8]
```diff
@@ -44,6 +55,18 @@ def create_field_history(self, entity, field, clock, value):
         #  self.add(history)
```
Comment (fabiogibson) [inline Platform.App/python-template/core/temporal/session.py]: Please remove commented code.

File: Platform.App/python-template/core/temporal/session.py [hunk 9]
```diff
@@ -44,6 +55,18 @@ def create_field_history(self, entity, field, clock, value):
...
+    def create_field_history_bulk(self, entity, field, clock, value):
+        """creates a new entity field history.
+        """
+        history = entity._history[field]()
+        history.rid = uuid4()
+        history.entity_id = entity.rid
+        history.clock_id = clock.rid
+        history.ticks = pg_extras.NumericRange(clock.ticks+1)
+        history.value = getattr(entity, field)
+        #  self.add(history)
```
Comment (fabiogibson) [inline Platform.App/python-template/core/temporal/session.py]: Please remove commented code.

File: Platform.App/python-template/model/persistence.py (hunk 5 above)
Comment (fabiogibson) [inline Platform.App/python-template/model/persistence.py]: Please remove commented code.

Comment (diogenesjcsilva) [reply Platform.App/python-template/model/persistence.py]: Removed commented code.

File: Platform.App/python-template/core/temporal/session.py (hunk 9 above)
Comment (diogenesjcsilva) [reply Platform.App/python-template/core/temporal/session.py]: Removed commented code.

File: Platform.App/python-template/core/temporal/session.py (hunk 8 above)
Comment (diogenesjcsilva) [reply Platform.App/python-template/core/temporal/session.py]: Removed commented code.

File: Platform.App/python-template/core/temporal/session.py (hunk 7 above)
Comment (diogenesjcsilva) [reply Platform.App/python-template/core/temporal/session.py]: Removed commented code.

File: Platform.App/python-template/core/temporal/listeners.py (hunk 6 above)
Comment (diogenesjcsilva) [reply Platform.App/python-template/core/temporal/listeners.py]: Moved helper to utils.py.

File: Platform.App/python-template/model/persistence.py (hunk 5 above)
Comment (diogenesjcsilva) [reply Platform.App/python-template/model/persistence.py]: Removed commented code.

File: Platform.App/python-template/model/persistence.py (hunk 4 above)
Comment (diogenesjcsilva) [reply Platform.App/python-template/model/persistence.py]: Removed unnecessary checks.

File: Platform.App/python-template/core/temporal/listeners.py (hunk 3 above)
Comment (diogenesjcsilva) [reply Platform.App/python-template/core/temporal/listeners.py]: Replaced commits with flushes, and added only one commit at the end.

File: Platform.App/python-template/app/blueprints/mapping.py (hunk 1 above)
Comment (diogenesjcsilva) [reply Platform.App/python-template/app/blueprints/mapping.py]: Solved.

File: Platform.App/python-template/core/temporal/listeners.py (hunk 2 above)
Comment (diogenesjcsilva) [reply Platform.App/python-template/core/temporal/listeners.py]: Replaced commits with flushes, and added only one commit at the end.

**Pull request merged**